ndonnx/_core/_default.py

```diff
@@ -0,0 +1,594 @@
+# Copyright (c) QuantCo 2023-2024
+# SPDX-License-Identifier: BSD-3-Clause
+
+from __future__ import annotations
+
+from collections.abc import Iterable, Sequence
+from typing import Literal
+
+import numpy as np
+
+import ndonnx as ndx
+import ndonnx._data_types as dtypes
+import ndonnx._opset_extensions as opx
+
+from ._utils import (
+    from_corearray,
+)
+
+
+class OperationsBlock:
+    """Interface for data types to implement top-level functions exported by ndonnx."""
+
+    # elementwise.py
+
+    def abs(self, x) -> ndx.Array:
+        return NotImplemented
+
+    def acos(self, x) -> ndx.Array:
+        return NotImplemented
+
+    def acosh(self, x) -> ndx.Array:
+        return NotImplemented
+
+    def add(self, x, y) -> ndx.Array:
+        return NotImplemented
+
+    def asin(self, x) -> ndx.Array:
+        return NotImplemented
+
+    def asinh(self, x) -> ndx.Array:
+        return NotImplemented
+
+    def atan(self, x) -> ndx.Array:
+        return NotImplemented
+
+    def atan2(self, y, x) -> ndx.Array:
+        return ndx.atan(ndx.divide(y, x))
```

Comment:
Is this meant as a class with actual "defaults" or is the only real default that we want `NotImplemented`? I feel the second is more reasonable.

Reply:
I agree, I'm going to break out the default shape implementations.